### PR TITLE
ESLint sharable config

### PR DESCRIPTION
**Pull request by valeriangalliat**
This is an attempt to factorize napi's ESLint rules against [semistandard](https://github.com/Flet/semistandard) (since a lot of rules were common to this style). `eslintrc.json` is roughly napi's `.eslintrc` with all rules already defined in semistandard removed so it's more readable and based on something more standard, documented and maintained in the JS community.

Though, semistandard is more strict on some points, and this is a good thing IMO (and we also had some rules disabled with a `TODO` marker that are now enabled), but this breaks linting on napi, so I included a `legacy` package to keep linting napi without changing the code. But for newer packages (I especially have in mind all integration microservices/libraries), I'd rather stick to the main config.

I removed the rules that are common with semistandard style from the readme too to keep it more concise, and also removed some rules we don't use anymore, like aligning of `=` and `:`.

Since we moved to Node 4 I also advised to use template strings instead of old-fashion `[].join`, and prefer arrow functions for lexical `this` binding over `self` (even if `self` is still supported and its consistency enforced). I also removed `async`/`step` stuff and recommended promises instead.

---

We really need a `eslint-config-busbud` to avoid copy/pasting the `.eslintrc` in all integration microservices/libraries repos (~15 of them right now and this is just the beginning). The `standard` factorization is only a proposal that we discussed internally in the integrations team, and that I want to discuss more widely here.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,66 +1,22 @@
...
+## Rules not (yet) enforced by ESLint
 
-var joined_strings = strings.join(',');
+### Basics
 
-while (true) {
-  console.log(joined_strings);
-}
-```
+ - 80 characters per line
```
Comment (alexcarruthers) [inline README.md]: this one.... we don't necessarily follow. @Chris911 and i have been setting our IDEs to 120 and trying to follow it as much as possible, but not strictly.

Comment (dustinblackman) [inline README.md]: I agree, I'm not a fan of character caps at all honestly. I just to go by my own judgement instead.

Comment (smaspe) [inline README.md]: 80? what is this 1997?

Comment (Chris911) [inline README.md]: 120 is the new 80.

Comment (valeriangalliat) [inline README.md]: Well this was in the original style guide, not sure to keep it or not. But I feel the same as you all about this rule.

Personally I try to keep the line length below 80 most of the times, but when it makes sense I can go above. The reason is while we have big screens, I quite never code fullscreen, and my splits are often round 80 chars width. But for sure no hard rule.

**Comment by valeriangalliat** [comment]
The rules that semistandard adds to napi current config:

```
> R.difference(Object.keys(st.rules), Object.keys(napi.rules)).map(k => [k, st.rules[k]]).forEach(x => co
[ 'accessor-pairs', 2 ]
[ 'arrow-spacing', [ 2, { before: true, after: true } ] ]
[ 'block-spacing', [ 2, 'always' ] ]
[ 'constructor-super', 2 ]
[ 'dot-location', [ 2, 'property' ] ]
[ 'generator-star-spacing',
  [ 2, { before: true, after: true } ] ]
[ 'no-class-assign', 2 ]
[ 'no-const-assign', 2 ]
[ 'no-dupe-args', 2 ]
[ 'no-dupe-class-members', 2 ]
[ 'no-duplicate-case', 2 ]
[ 'no-extra-bind', 2 ]
[ 'no-irregular-whitespace', 2 ]
[ 'no-multiple-empty-lines', [ 2, { max: 1 } ] ]
[ 'no-this-before-super', 2 ]
[ 'no-throw-literal', 2 ]
[ 'no-unexpected-multiline', 2 ]
[ 'no-unneeded-ternary', [ 2, { defaultAssignment: false } ] ]
[ 'no-useless-call', 2 ]
[ 'one-var', [ 2, { initialized: 'never' } ] ]
[ 'operator-linebreak',
  [ 2, 'after', { overrides: [Object] } ] ]
[ 'padded-blocks', [ 2, 'never' ] ]
[ 'radix', 2 ]
[ 'space-before-keywords', [ 2, 'always' ] ]
[ 'wrap-iife', [ 2, 'any' ] ]
[ 'standard/object-curly-even-spacing', [ 2, 'either' ] ]
[ 'standard/array-bracket-even-spacing', [ 2, 'either' ] ]
[ 'standard/computed-property-even-spacing', [ 2, 'even' ] ]
```

**Comment by valeriangalliat** [comment]
With @killix +1 on Trello, @alexcarruthers +1 on Flowdock, and @dustinblackman +1 face to face (why everyone is shy to post on the PR ? :cry:), I'm merging this PR. :guitar:

Ping me if anything's wrong with this ESLint refactoring, but the legacy one should be the same as napi's current config, and the main one have what's `TODO` in napi enabled plus semistandard additional checks (mostly ES6 stuff).

**Pull request merged**